sympy/physics/units/quantities.py

```diff
@@ -89,7 +89,7 @@ def _eval_is_constant(self):
         return self.scale_factor.is_constant()
 
     def _eval_Abs(self):
-        return self.__class__(self.name, self.dimension,
+        return Quantity(self.name, self.dimension,
```

Comment:
`.func( )` usually works as the class constructor, with some few exception when a parameter in the class constructor is curried. For example, logical predicates are like that: `Q.positive(x)` is constructed by `AppliedPredicate(Q.positive, x)` but you should use it as if it has only one parameter. So if you want to rebuild it with `y`, for instance, just call `Q.positive(x).func(y)`